weatherreport.cpp

```diff
@@ -62,20 +62,46 @@ namespace WeatherSpace
 
     void TestHighPrecipitation()
     {
-        // This instance of stub needs to be different-
-        // to give high precipitation (>60) and low wind-speed (<50)
         SensorStub sensor;
+        string report = Report(sensor);
+        // Test for high precipitation (>60) and low wind speed (<50)
+        assert(report.find("rain") != string::npos);
+    }
+
+    void TestSunnyDay()
+    {
+        class SunnySensorStub : public IWeatherSensor {
+            int Humidity() const override { return 30; }
+            int Precipitation() const override { return 10; }
+            double TemperatureInC() const override { return 20; }
+            int WindSpeedKMPH() const override { return 10; }
+        };
+        
+        SunnySensorStub sensor;
+        string report = Report(sensor);
+        assert(report == "Sunny Day");
+    }
 
-        // strengthen the assert to expose the bug
-        // (function returns Sunny day, it should predict rain)
+    void TestPartlyCloudy()
+    {
+        class PartlyCloudySensorStub : public IWeatherSensor {
```

Comment:
Nice and clear tests!
A choice: You have chosen to make different classes to represent different weather conditions. Another option would have been to make one Stub class and accept the values to be returned in the constructor. That way you would have multiple objects instead of multiple classes.
Which way is better? Please lookup "Property based tests" to make up your mind